crates/re_viewer_context/src/space_view/space_view_class_impl.rs

```diff
@@ -1,21 +1,31 @@
-use crate::{Scene, SpaceViewClass, SpaceViewClassName, SpaceViewState, ViewerContext};
+use crate::{
+    Scene, SceneContext, ScenePartCollection, SpaceViewClass, SpaceViewClassName, SpaceViewState,
+    ViewerContext,
+};
 
-use super::scene_element_list::SceneElementListConversionError;
+use super::scene::TypedScene;
 
-/// Utility for implementing [`SpaceViewClass`] with concrete [`SpaceViewState`] and [`crate::SceneElement`] type.
+/// Utility for implementing [`SpaceViewClass`] with concrete [`SpaceViewState`] and [`crate::ScenePart`] type.
 ///
 /// Each Space View in the viewer's viewport has a single class assigned immutable at its creation time.
 /// The class defines all aspects of its behavior.
 /// It determines which entities are queried, how they are rendered, and how the user can interact with them.
-pub trait SpaceViewClassImpl {
+pub trait SpaceViewClassImpl: std::marker::Sized {
```

Comment:
I think we should consider swapping the names of `SpaceViewClassImpl` and `SpaceViewClass` -- this trait is the important one, which is user-visible and must be implemented to instantiate a SpaceViewClass... it's the other trait that is actually an implementation detail of the SpaceView engine but I suspect doesn't even need to be public outside of this crate making it much more suited to be called impl.